### ISSUE TITLE
logoutRedirect always shows account picker when called with an account and idTokenClaim.login_hint is set

### ISSUE THREAD

**Issue by bj00rn** [AzureAD/microsoft-authentication-library-for-js#4535]
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Core Library Version

2.21.0

### Wrapper Library

MSAL React (@azure/msal-react)

### Wrapper Library Version

1.2.0

### Description

Promptless logoutRedirect (https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/logout.md#promptless-logout) is not working.

Account picker is shown even if a valid account is passed that has the `idTokenClaims.login_hint` set

### Msal Logs

[Fri, 18 Feb 2022 13:51:03 GMT] : @azure/msal-browser@2.21.0 : Info - Emitting event: msal:logoutStart
[Fri, 18 Feb 2022 13:51:03 GMT] : @azure/msal-react@1.2.0 : Info - MsalProvider - msal:logoutStart results in setting inProgress from none to logout
[Fri, 18 Feb 2022 13:51:03 GMT] : @azure/msal-browser@2.21.0 : Info - Emitting event: msal:logoutSuccess
[Fri, 18 Feb 2022 13:51:08 GMT] : @azure/msal-browser@2.21.0 : Info - Emitting event: msal:handleRedirectStart
[Fri, 18 Feb 2022 13:51:08 GMT] : @azure/msal-react@1.2.0 : Info - MsalProvider - msal:handleRedirectStart results in setting inProgress from startup to handleRedirect
[Fri, 18 Feb 2022 13:51:08 GMT] : @azure/msal-browser@2.21.0 : Info - Emitting event: msal:handleRedirectEnd
[Fri, 18 Feb 2022 13:51:08 GMT] : @azure/msal-react@1.2.0 : Info - MsalProvider - no account changes
[Fri, 18 Feb 2022 13:51:08 GMT] : @azure/msal-react@1.2.0 : Info - MsalProvider - msal:handleRedirectEnd results in setting inProgress from handleRedirect to none

### MSAL Configuration

```javascript
{
    auth: {
        authority: 'https://login.microsoftonline.com/common',
        clientId: 'xxxxx-xxxx-x-x-x-x-x-x-x-xxxxxxx',
        redirectUri: 'https://xxx.xxx.com',
    },
    cache: {
        cacheLocation: 'localStorage',
        storeAuthStateInCookie: false,
        secureCookies: true,
    },
    system: {
        loggerOptions: {
            loggerCallback: (level: LogLevel, message: string, containsPii: boolean): void => {
                if (containsPii) {
                    return
                }
                console.log(message)
            },
            piiLoggingEnabled: false,
        },
    }
}
```

### Relevant Code Snippets

```javascript
function logoutMsal(accountId?: string) {
    const endSessionRequest: EndSessionRequest = {
        // has to be an absolute url hence the window.location...
        postLogoutRedirectUri: `${window.location.origin}/login`,
    }
    if (accountId) {
        const logoutAccount = _msalInstance.getAccountByHomeId(accountId)
        if (logoutAccount) {
            endSessionRequest.account = logoutAccount
        }
    }
    return _msalInstance.logoutRedirect(endSessionRequest)
}
```

### Reproduction Steps

1. make sure idTokenClaims.login_hint set for account
2. pass account to logoutRedirect
3. Browser is redirected to accountPicker prompting the user to select which account to sign out of

### Expected Behavior

According to documentation, promptless logout should be performed without need for user interaction when an account is passed to logoutRedirect and `account.idTokenClaims.login_hint` claim is present.

### Identity Provider

Azure AD / MSA

### Browsers Affected (Select all that apply)

Chrome

### Source

External (Customer)

**Comment by bj00rn** [comment]
https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/3559

**Comment by hectormmg** [comment]
@bj00rn it seems you are using version 2.21 of msal-browser. The logout hint support feature was shipped on the latest version, 2.22. Could you please upgrade your msal-browser version and let us know if your issue is resolved? We’ll make sure to update documentation to make it clear that this is a v2.22+ feature.

**Comment by bj00rn** [comment]
> @bj00rn it seems you are using version 2.21 of msal-browser. The logout hint support feature was shipped on the latest version, 2.22. Could you please upgrade your msal-browser version and let us know if your issue is resolved? We’ll make sure to update documentation to make it clear that this is a v2.22+ feature.

Ok, i see. Seems it's still an issue with 2.21 i'm afraid. Still getting the account picker. I'm passing my account to logoutRedirect as shown below.


![image](https://user-images.githubusercontent.com/10961167/155089593-be022789-8295-4715-bc33-980d2cc2d0da.png)

![image](https://user-images.githubusercontent.com/10961167/155090005-210a533b-5fea-49d1-8797-9850283620ef.png)

**Comment by bj00rn** [comment]
> @bj00rn it seems you are using version 2.21 of msal-browser. The logout hint support feature was shipped on the latest version, 2.22. Could you please upgrade your msal-browser version and let us know if your issue is resolved? We’ll make sure to update documentation to make it clear that this is a v2.22+ feature.

![image](https://user-images.githubusercontent.com/10961167/155096301-0a08c719-556c-448c-b645-8a10a67908e8.png)

Edit: it's working if I set both `account` and `logoutHint` in `EndSessionRequest` is this a requirement? Maybe the documentation should be updated if this is the case. Setting only `accont` in `EndSessionRequest` still shows the account picker.

**Comment by bj00rn** [comment]
> @bj00rn it seems you are using version 2.21 of msal-browser. The logout hint support feature was shipped on the latest version, 2.22. Could you please upgrade your msal-browser version and let us know if your issue is resolved? We’ll make sure to update documentation to make it clear that this is a v2.22+ feature.

Thanks! Please do 🌹 .

Confirmed working with `v2.22`

Closing
